sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubClient.cs

```diff
@@ -0,0 +1,282 @@
+        ///   connection string.  For example, ";EntityPath=telemetry-hub".
+        ///
+        ///   If you have defined a shared access policy directly on the Event Hub itself, then copying the connection string from that
+        ///   Event Hub will result in a connection string that contains the path.
+        /// </remarks>
+        ///
+        public EventHubClient(string                connectionString,
+                              EventHubClientOptions clientOptions)
+        {
+            EventHubPath = "THIS WOULD BE PARSED FROM THE CONNECTION STRING";
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventHubClient"/> class.
+        /// </summary>
+        ///
+        /// <param name="host">The fully qualified host name for the Event Hubs namespace.  This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.</param>
+        /// <param name="eventHubPath">The path of the specific Event Hub to connect the client to.</param>
+        /// <param name="credential">The Azure managed identity credential to use for authorization.  Access controls may be specified by the Event Hubs namespace or the requeseted Event Hub, depending on Azure configuration.</param>
+        /// <param name="clientOptions">A set of options to apply when configuring the client.</param>
+        ///
+        public EventHubClient(string                  host,
+                              string                  eventHubPath,
+                              TokenCredential         credential,
+                              EventHubClientOptions   clientOptions = default)
+        {
+            Guard.ArgumentNotNull(nameof(host), host);
+            Guard.ArgumentNotNullOrEmpty(nameof(eventHubPath), eventHubPath);
+            Guard.ArgumentNotNull(nameof(credential), credential);
+
+            EventHubPath = eventHubPath;
+            Credential = credential;
+            ClientOptions = clientOptions;
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventHubClient"/> class.
+        /// </summary>
+        ///
+        protected EventHubClient()
+        {
+        }
+
+        /// <summary>
+        ///   Retrieves information about an Event Hub, including the number of partitions present
+        ///   and their identifiers.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request for cancelling the operation.</param>
+        ///
+        /// <returns>The set of information for the Event Hub that this client is associated with.</returns>
+        ///
+        public virtual Task<EventHubProperties> GetPropertiesAsync(CancellationToken cancellationToken = default) =>
+            Task.FromResult(new EventHubProperties("/sample-path", DateTime.UtcNow.AddDays(-5), 3, new[] { "one", "two", "three" }, DateTime.UtcNow));
```

Comment:
Is this here as an example?

Reply:
Yes.   Trust no implementation.   This is about the shape of the API and doc comments only.   Any implementation is just there to ensure that it compiles and that I could do some initial sampling.